### PR TITLE
Setup to enable Rebus header to cloudevent attributes (and back)

### PR DESCRIPTION
**Pull request by Powdor42**

### CODE REVIEW

File: Rebus/CloudEventify.Rebus/RebusHeader2CloudAttributeMap.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    };
+
+    public string FromCloudAttribute(string cloudAttributeName)
+    {
+        return Rebus2CloudMap.First(p => p.Value.Equals(cloudAttributeName, StringComparison.OrdinalIgnoreCase)).Key; ;
+    }
+
+    public string ToCloudAttributeName(string rebusHeaderName)
+    {
+        return Rebus2CloudMap.First(p=> p.Key.Equals(rebusHeaderName, StringComparison.Ordinal)).Value;
```
Comment (mstepanova-xpirit) [inline Rebus/CloudEventify.Rebus/RebusHeader2CloudAttributeMap.cs]: Do you really need such a complicated search? Rebus2CloudMap is a Dictionary, so key is unique 
If you are absolutely sure that rebusHeaderName always exists isn't it better use just Rebus2CloudMap[rebusHeaderName]? Also keep in mind that if key rebusHeaderName doesn't exist for any reason you'll get and exception

File: Rebus/CloudEventify.Rebus/HeaderMap.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+{
+    private HeaderMap() { }
+
+    private static readonly object _lock = new object ();
+    private static Map<string,string> _instance = null!;
+    
+    public static Map<string,string> Instance
+    {
+        get
+        {
```
Comment (mstepanova-xpirit) [inline Rebus/CloudEventify.Rebus/HeaderMap.cs]: Its been locked every time its requested. Here we need either double check lock pattern or I'd even transfer its initialization to declaration as its always thread safe

File: CloudEventify/ITypeMap.cs [hunk 3]
```diff
@@ -1,6 +1,6 @@
 namespace CloudEventify;
```
Comment (mstepanova-xpirit) [inline CloudEventify/ITypeMap.cs:2]: Whats the difference between this interface and IDictionary<T,T>, doesn't it have the same method?

Comment (Powdor42) [reply CloudEventify/ITypeMap.cs:2]: I cannot change this. it is not a back and forth map (it has 2 different mappings).. this is existing code to which I only renamed the object..

File: Rebus/CloudEventify.Rebus/HeaderMap.cs (hunk 2 above)
Comment (Powdor42) [reply Rebus/CloudEventify.Rebus/HeaderMap.cs]: fixed

**Review by mstepanova-xpirit — APPROVED** [review]

**Comment by riezebosch** [comment]
Missing message-id fixed by #7, other headers are on-hold for now.

**Pull request closed without merging**